File: velocity/src/main/java/me/devsnox/pingcounter/velocity/PingListener.java
package me.devsnox.pingcounter.velocity;

import com.velocitypowered.api.event.Subscribe;
import com.velocitypowered.api.event.proxy.ProxyPingEvent;
import me.devsnox.pingcounter.core.api.CounterEngine;

public class PingListener {

    private final CounterEngine engine;

    PingListener(final CounterEngine engine) {
        this.engine = engine;
    }

    @Subscribe
    public void onPing(final ProxyPingEvent event) {
        this.engine.addPing(event.getConnection().getRemoteAddress().getAddress());
    }
}

File: velocity/src/main/java/me/devsnox/pingcounter/velocity/PingListener.kt
package me.devsnox.pingcounter.velocity

import com.velocitypowered.api.event.Subscribe
import com.velocitypowered.api.event.proxy.ProxyPingEvent
import me.devsnox.pingcounter.core.api.CounterEngine

class PingListener internal constructor(private val engine: CounterEngine) {

    @Subscribe
    fun onPing(event: ProxyPingEvent) {
        this.engine.addPing(event.connection.remoteAddress.address)
    }
}
